docs/overview.adoc

```diff
@@ -0,0 +1,331 @@
+= Architecture
+
+== Resource types
+
+One concept this library is built around are so-called resource types.
+In the following sections we will introduce the importance of them to the overall architecture and how they are to be used to implement the SoC (separation of concerns) and dry (don't repeat yourself) principles.
+
+=== Catch up -- resource types as defined by the JSON:API specification
+
+Within the JSON:API specification every resource is assigned a type, which is simply a single string.
+The naming of the type depends on decisions made when designing the domain model of the API.
+
+If your website publishes online articles and allows users to post comments under them, an article with a single comment may exist.
+Such case is shown in the following example.
+The JSON displays a single `article` resource with the `id` 42.
+Other typical article relationships, e.g. the authors, are omitted for brevity.
+However, it contains a list of comment references, containing a single reference.
+
+[source,json]
+----
+{
+  "data": {
+    "id": "42",
+    "type": "article",
+    "attributes": {
+      "text": "…"
+    },
+    "relationships": {
+      "comments": {
+        "data": [
+          {"id": 42, "type": "comment"}
+        ]
+      }
+    }
+  }
+}
+----
+
```

Comment:
Maybe it would be simpler to just use different IDs in the example. While reading the example-code I was questioning if this is a kind of reference. The text afterwards explains it, so its ok I guess. But using different IDs in the example and writing a text afterwards which explains that it is allowed to use the same ID for different types, would avoid confusion in the first place

Reply:
I've changed the ID of the example comment.